### ISSUE TITLE
Calamares doesn't keep boot flag in manual installation (UEFI)

### ISSUE THREAD

**Issue by deesnook** [calamares/calamares#998]
I did a clean install of Manjaro Gnome (v17.1.10) using Calamares and encountered the following problem: When doing a UEFI installation, Calamares does not retain the boot/efi flags set for the EFI partition. Thus, after installation and upon reboot there will be no bootable medium. You can work around this by partitioning the HDD with GParted and setting the necessary flags there, but with Calamares it won't work. You can also visualize the problem within Calamares: If you set the flags and continue, but then go back to the settings for the EFI partition before starting the installation, the flags you have just set are gone.

1. Start Calamares installation
2. Choose manual partitioning
3. Create an EFI partition (and the other partitions needed for a complete install)
4. Set boot/efi flags for the EFI partition
5. either: complete installation and reboot --> no bootable medium; or: within Calamares, go back to the settings of the EFI partition --> flags will be gone

Expected behavior
Calamares retains the boot flags

**Comment by adriaandegroot** [comment]
This is roughly #976 (but with UEFI instead of MBR). In general, *manual partition* means "I know what I'm doing" and Calamares won't do anything smart for you.

**But** I see you have step 4, and set the flags explicitly. Since you're using an older Calamares, some things have since changed:
 - within Calamares, the flag is preserved (it was a UI problem, which would then erase the flag when you leave the page again),
 - as for setting the flag on the disk itself, that will need testing; but I do remember that there was confusion between MBR flags and UEFI flags for boot (they are different bits). There's a closed issue for that somewhere.

**Comment by deesnook** [comment]
Thanks Adrian, I didn't know I was using an older version of Calamares, I installed the then current version of Manjaro and just assumed that the image contained the current version of Calamares too. So I understand the problem has been fixed now?

**Comment by adriaandegroot** [comment]
@deesnook There's "the current version of Calamares", and then there's "the version of Calamares that was current at the time the ISO you're using was created". The problem you're describing has been **partly** fixed (possibly wholly fixed -- that's what needs checking). If you're happy building software you might try the current version of Calamares (3.2.1) from that same ISO.

In any case, I'm leaving the issue open until it has been looked at at least once from a recent Manjaro ISO.

**Comment by philmmanjaro** [comment]
@deesnook: Please try at least our current stable release of Manjaro, which is **v17.1.11**. There we ship with **CAL v3.2**.

**Comment by deesnook** [comment]
@philmmanjaro yup, that's what I have planned. Will do as soon as I have the time for it

**Comment by adriaandegroot** [comment]
I've just checked on a recent ISO and a build of Calamares from master, and it keeps the flag there, and the partition is created with the right flags as well. So closing this as "already fixed".
